ckanext/stadtzhtheme/commands.py

```diff
@@ -165,6 +165,7 @@ def _cleanup_storage_files(self, context, resource_path):
                     resource_id = ''.join(root.split('/')[-2:]) + file
                     file_path = os.path.join(root, file)
                     try:
+                        logic.check_access('resource_show', context)
```

Comment:
You don't need to do this for every single file. You can check for it once at the start of the method.